src/librustc_hir/lang_items.rs

```diff
@@ -47,44 +47,86 @@ macro_rules! language_item_table {
             }
         }
 
+        #[derive(Debug, Copy, Clone, Hash)]
+        pub enum LangItemRecord {
+            /// The language item was found and mapped to its `DefId`
+            /// during collection.
+            DefId(DefId),
+            /// The language item was not found during collection.
+            Missing(LangItem),
+        }
+
+        impl LangItemRecord {
+            /// Requires that the `LangItem` was bound and returns the corresponding `DefId`.
+            pub fn require(self) -> Result<DefId, LangItem> {
+                match self {
+                    LangItemRecord::DefId(def_id) => Ok(def_id),
+                    LangItemRecord::Missing(item) => Err(item),
+                }
+            }
+
+            /// Checks the current `LangItemRecord` against the given `DefId`.
+            /// Returns true if and only if the language item was found and its `DefId`
+            /// is equal to the `DefId` provided.
+            pub fn is(self, other: DefId) -> bool {
```

Comment:
Did you consider `PartialEq<DefId>` instead of a custom method?

Reply:
Yeah I thought about it but decided it seemed a little opaque - do you think `PartialEq<DefId>` would be better?

Reply:
I'm mostly indifferent but also lean towards an inherent method. Just wondering if you had a more convincing argument. FWIW, I probably would have chosen a different name (`has_def_id`), but that's just an opinion.